SMLHelper/Options/Attributes/MenuAttribute.cs

```diff
@@ -0,0 +1,200 @@
+    ///                 Logger.Log(Logger.Level.Info, sliderChangedEventArgs.Value.ToString());
+    ///                 break;
+    ///             case ToggleChangedEventArgs toggleChangedEventArgs:
+    ///                 Logger.Log(Logger.Level.Info, toggleChangedEventArgs.Value.ToString());
+    ///                 break;
+    ///         }
+    ///      }
+    /// }
+    /// </code>
+    /// </example>
+    /// <seealso cref="ChoiceAttribute"/>
+    /// <seealso cref="OnChangeAttribute"/>
+    /// <seealso cref="KeybindAttribute"/>
+    /// <seealso cref="SliderAttribute"/>
+    /// <seealso cref="ToggleAttribute"/>
+    /// <seealso cref="ButtonAttribute"/>
+    /// <seealso cref="ModOptions"/>
+    /// <seealso cref="ConfigFile"/>
+    /// <seealso cref="Logger"/>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+    public sealed class MenuAttribute : Attribute
+    {
+        /// <summary>
+        /// Specifies after which events the config file should be saved to disk automatically.
+        /// </summary>
+        /// <remarks>
+        /// This enumeration has a <see cref="FlagsAttribute"/> that allows a bitwise combination of its member values.
+        /// </remarks>
+        [Flags]
+        public enum SaveEvents : byte
+        {
+            /// <summary>
+            /// Never automatically save.
+            /// </summary>
+            None = 0,
+            /// <summary>
+            /// Save whenever any value is changed.
+            /// </summary>
+            ChangeValue = 1,
+            /// <summary>
+            /// Save when the player saves the game.
+            /// </summary>
+            SaveGame = 2,
+            /// <summary>
+            /// Save when the player quits the game.
+            /// </summary>
+            QuitGame = 4
+        }
+
+        /// <summary>
+        /// Specifies after which events the config file should be loaded from disk automatically.
+        /// </summary>
+        /// <remarks>
+        /// This enumeration has a <see cref="FlagsAttribute"/> that allows a bitwise combination of its member values.
+        /// </remarks>
+        [Flags]
+        public enum LoadEvents : byte
+        {
+            /// <summary>
+            /// Never automatically load.
+            /// </summary>
+            None = 0,
+            /// <summary>
+            /// Load when the menu is registered to SMLHelper via <see cref="Handlers.OptionsPanelHandler.RegisterModOptions{T}"/>.
+            /// </summary>
+            /// <remarks>
+            /// In normal usage, this option is equivalent to loading when the game is launched.
+            /// </remarks>
+            MenuRegistered = 1,
+            /// <summary>
+            /// Load when the menu is opened by the player.
+            /// </summary>
+            /// <remarks>
+            /// Useful for allowing players to edit their config files manually without restarting the game for their changes to take effect.
```

Comment:
Can I get either an option to have this load on register as well when set to load on menu open or another option that does both as not everyone opens the menu when they go to start a new game if they have just changed the config file.

Reply:
It's a flags enum. You can do both by specifying to do both ;)

Reply:
https://github.com/SubnauticaModding/SMLHelper/pull/173#discussion_r484490882
^ as I said to you here ;)

Reply:
I am an idiot.... proceed with yourself and ignore that i ever said this